src/frontends/pytorch/src/transforms/quantize_dequantize_replacer.cpp

```diff
@@ -0,0 +1,96 @@
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "utils.hpp"
+#include "utils_quantize.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+using namespace ov::opset10;
+
+QuantizeDequantizeReplacer::QuantizeDequantizeReplacer() {
+    auto dequantize_node = ov::pass::pattern::wrap_type<ov::frontend::pytorch::QuantizedPtNode>();
+
+    ov::matcher_pass_callback callback = [](ov::pass::pattern::Matcher& m) {
+        auto dequantize_node = cast_quantized_fw_node(m.get_match_root(), "aten::dequantize");
+        if (!dequantize_node)
+            return false;
+
+        ov::pass::NodeRegistry rg;
+        std::shared_ptr<ov::Node> quantize_input, dequantize_input, scale, zero_point, axis;
+
+        auto quantize_node = dequantize_node->input_value(0).get_node_shared_ptr();
+        if (auto quantize_per_tensor = cast_quantized_fw_node(quantize_node, "aten::quantize_per_tensor")) {
+            quantize_input = quantize_per_tensor->input_value(0).get_node_shared_ptr();
+            scale = quantize_per_tensor->input_value(1).get_node_shared_ptr();
+            zero_point = quantize_per_tensor->input_value(2).get_node_shared_ptr();
+            auto const dtype_node = quantize_per_tensor->input_value(3).get_node_shared_ptr();
+            auto const const_dtype_node = std::dynamic_pointer_cast<opset10::Constant>(dtype_node);
+            auto const dtype = convert_dtype(const_dtype_node->cast_vector<int64_t>().at(0));
+
+            // Quantize
+            const auto scale_convert = rg.make<opset10::ConvertLike>(scale, quantize_input);
+            const auto zero_point_convert = rg.make<opset10::ConvertLike>(zero_point, quantize_input);
+            const auto scaled_input = rg.make<opset10::Divide>(quantize_input, scale_convert);
+            const auto scaled_input_with_zero_pt = rg.make<opset10::Add>(scaled_input, zero_point_convert);
+            const auto quantized_input =
+                rg.make<opset10::Round>(scaled_input_with_zero_pt, opset10::Round::RoundMode::HALF_TO_EVEN);
+
+            if (dtype == element::u8) {
+                const auto clamp = rg.make<opset10::Clamp>(quantized_input,
+                                                           std::numeric_limits<unsigned char>::lowest(),
+                                                           std::numeric_limits<unsigned char>::max());
+                dequantize_input = rg.make<opset10::Convert>(clamp, element::u8);
+            } else if (dtype == element::i8) {
+                const auto clamp = rg.make<opset10::Clamp>(quantized_input,
+                                                           std::numeric_limits<char>::lowest(),
+                                                           std::numeric_limits<char>::max());
+                dequantize_input = rg.make<opset10::Convert>(clamp, element::i8);
+            } else {
+                dequantize_input = rg.make<opset10::Convert>(quantized_input, element::i32);
```

Comment:
what about float16 quantization?

Reply:
Not sure about this, since both quantize_per_tensor and quantize_per_channel don't mention float16 dtype

Reply:
I decided to add support for this type in quantize operations, so this case is now covered!